File: java/src/main/java/mx/iteso/adapter/impl/GoogleDocAdapter.java
package mx.iteso.adapter.impl;


import mx.iteso.adapter.IGoogleDoc;
import mx.iteso.adapter.IWordDocument;
import mx.iteso.adapter.MSLicense;
import mx.iteso.adapter.media.Format;
import mx.iteso.adapter.media.Image;


/**
 * Created by Margarita Jauregui on 11/11/19.
 */
public class GoogleDocAdapter implements IWordDocument {
    /**
     * IGoogleDoc object.
     */
    private IGoogleDoc googleDoc;
    /**
     * Version number.
     */
    private float msOfficeVersion;
    /**
     * License number.
     */
    private MSLicense msLicense;
    /**
     * Format.
     */
    private Format format;

    /**
     * Class constructor.
     * @param googleDoc target
     */
    public GoogleDocAdapter(IGoogleDoc googleDoc) {
        this.googleDoc = googleDoc;
    }

    /**
     * Format getter.
     * @return format
     */
    public Format getFormat() {
        return (Format) this.googleDoc.getStyle();
    }

    /**
     * Format setter.
     * @param format new format
     */
    public void setFormat(Format format) {
        this.format = format;
    }

    /**
     * Image getter.
     * @return Image
     */
    public Image getBackground() {
        return this.googleDoc.getBackground();
    }

    /**
     * MSOfficeVersion setter.
     * @param msOfficeVersion number
     */
    public void setMSOfficeVersion(final float msOfficeVersion) {
        return;
    }

    /**
     * MSLicense getter.
     * @return
     */
    public MSLicense getLicense() {
        return null;
    }

    /**
     * Getter for permission.
     * @param msLicense number
     * @return
     */
    public boolean restrictEditIfLicenseIsInvalid(final MSLicense msLicense) {
        return false;
    }
}

File: csharp/adapter-practice/GoogleDocAdapter.cs


namespace adapter_practice
{
    public class GoogleDocAdapter:IWordDocument
    {
    
    private readonly IGoogleDoc _googleDoc;
   
    private readonly MsLicense _msLicense;
   
    private float _msVersion;
    
    public GoogleDocAdapter(IGoogleDoc googledoc)
    {
        this._googleDoc = googledoc;
        this._msLicense = new MsLicense("Google license");
    }
    public Format GetFormat()
    {
        return new Format(_googleDoc.GetFont());
    }
    public Img GetBackground()
    {
        BackgroundImg backgroundImage = _googleDoc.GetBackground();
        return new Img(backgroundImage.GetImage().GetUrl());
    }
    public void SetMOfficeVersion(float msOfficeVersion)
    {
        this._msVersion = msOfficeVersion;
    }
    public  float GetMsOfficeVersion()
    {
        return this._msVersion;
    }
    public MsLicense GetLicense()
    {
        return this._msLicense;
    }
    public bool RestrictEditIfLicenseIsInvalid(MsLicense ms)
    {
        return this._msLicense.GetLicense() == ms.GetLicense();
    }

    }
}
